src/data/nat/modeq.lean

```diff
@@ -154,6 +168,79 @@ lemma le_of_lt_add (h1 : a ≡ b [MOD m]) (h2 : a < b + m) : a ≤ b :=
 lemma add_le_of_lt (h1 : a ≡ b [MOD m]) (h2 : a < b) : a + m ≤ b :=
 le_of_lt_add (add_modeq_right.trans h1) (add_lt_add_right h2 m)
 
+lemma dvd_iff_of_modeq_of_dvd {a b d m : ℕ} (h : a ≡ b [MOD m]) (hdm : d ∣ m) :
+  d ∣ a ↔ d ∣ b :=
+begin
+  simp only [←modeq_zero_iff_dvd],
+  have habd := modeq.modeq_of_dvd hdm h,
+  exact ⟨(modeq.symm habd).trans, modeq.trans habd⟩,
+end
```

Comment:
We can use some dot notation to make it somewhat easier to read:
```suggestion
begin
  simp only [←modeq_zero_iff_dvd],
  replace h := h.modeq_of_dvd hdm,
  exact ⟨h.symm.trans, h.trans⟩,
end
```
(Also I used `replace` in case you like it.)

Reply:
Very nice, thanks!